crates/llvm-context/src/polkavm/context/function/runtime/immutable_data_load.rs

```diff
@@ -37,6 +39,15 @@ where
         context.set_current_function(runtime::FUNCTION_LOAD_IMMUTABLE_DATA)?;
         context.set_basic_block(context.current_function().borrow().entry_block());
 
+        if context.debug_info().is_some() {
+            context.builder().unset_current_debug_location();
+            let func_scope = context
+                .set_current_function_debug_info(runtime::FUNCTION_LOAD_IMMUTABLE_DATA, 0)?
+                .as_debug_info_scope();
+            context.debug_info().unwrap().push_scope(func_scope);
+            context.set_debug_location(0, 0, Some(func_scope))?;
+        }
+
```

Comment:
This pairs with `set_current_function`? Setting the current function implies adjusting the debug info location. So it should be done in `fn set_current_function`.

This also reduces the potential to forget about it. For example, in the EVMLA codegen it is missing.

Reply:
The EVMLA will be done separately, this PR is concerned with the LLVM part of the pipeline.

Reply:
Well, EVMLA code is lowered to LLVM IR by mostly the same backend code. But EVMLA is of lower priority than YUL. As long as the EVMLA pipeline does not break as a result of this PR, I'm ok with doing it later.